### PR TITLE
Add extP2P and OutIdx validity check

### PR DESCRIPTION
**Pull request by devcareer0**
PR for implementation of 
https://freedcamp.com/view/2956206/tasks/38925962
and
https://freedcamp.com/view/2956206/tasks/40300409

- mnode-config.cpp / .h :
	- isOutIdxValid() check function
	-  extP2P parameter addition
- mnode-masternode.cpp / .h and mnode-rpc.cpp
	- extP2P parameter addition
- mn_common.py : added +extP2P parameter creation to create_masternode_config function
- mn_governance__custom_test.py: added test case for extP2P check

**Test results:**
Both extP2P and isOutIdxValid() tested. _mn_common.py_ contains extP2P param creation.
extP2P tests can be found in mn_governance__custom_test.py and isOutIdxValid() tests were performed on local feature branch with the modifications of in _mn_common.py_ , (should not be in production code). See a code snippet below how can you reproduce (modifying _create_masternode_conf_ function)

```
def create_masternode_conf(name, n, dirname, txid, vin, private_key, mn_port, **num**):
    datadir = os.path.join(dirname, "node"+str(n))
    if not os.path.isdir(datadir):
        os.makedirs(datadir)    
    regtestdir = os.path.join(datadir, "regtest")
    if not os.path.isdir(regtestdir):
        os.makedirs(regtestdir)    
    
    cfg_file = os.path.join(regtestdir, "masternode.conf")
    
    config = {}
    if os.path.isfile(cfg_file):
        with open(cfg_file) as json_file:  
            config = json.load(json_file)

    config[name] = {}
    config[name]["mnAddress"] = "127.0.0.1:" + str(mn_port)
    config[name]["mnPrivKey"] = str(private_key)
    config[name]["txid"] = str(txid)
    config[name]["outIndex"] = str(vin)
    **if num ==5:
        config[name]["outIndex"] = "100000" # OK
    elif num ==6:
        config[name]["outIndex"] = "1000000" # Still OK
    elif num ==7:
        config[name]["outIndex"] = "sdadsad" # Error: Failed to parse outIndex string. Alias: mn7 (outIndex)
    elif num == 8:
        config[name]["outIndex"] = "1000001"** # Error: Error reading masternode configuration file: Failed to parse outIndex string. #Value shall be between 0 and 1000000
#Alias: mn8 (outIndex)
    
    config[name]["extAddress"] = "127.0.0.1:" + str(random.randint(2000, 5000))
    config[name]["extP2P"] = "127.0.0.1:" + str(random.randint(22000, 25000))
    config[name]["extKey"] = ''.join(random.sample(private_key,len(private_key)))
    config[name]["extCfg"] = {}
    config[name]["extCfg"]["param1"] = str(random.randint(0, 9))
    config[name]["extCfg"]["param2"] = str(random.randint(0, 9))

    with open(cfg_file, 'w') as f:
        json.dump(config, f, indent=4)
    return datadir
```

### CODE REVIEW

**Review by a-ok123 — CHANGES_REQUESTED** [review]
Test for masternode config is included into mn_governance__custom_test.py
Please either rename it as something more tests specific (ex: mn_bugs.py), or - move masternode conf tests into mn_main.py

**Comment by devcareer0** [comment]
> mn_bugs.py

Ok. Renamed custom file.

**Comment by devcareer0** [comment]
Modified according to our discussion.

**Review by a-ok123 — APPROVED** [review]

File: src/mnode/mnode-config.cpp [hunk 1]
```diff
@@ -23,6 +23,33 @@ using json = nlohmann::json;
     }
 */
 
+bool isOutIdxValid(std::string& outIdx, std::string alias, std::string& strErr)
+{
+    bool retVal = false;
+    char* p = nullptr;
+    long converted = strtol(outIdx.c_str(), &p, 10);
+
+    if(*p != '\0')
```
Comment (akobrin1) [inline src/mnode/mnode-config.cpp:32]: you must check p also for nullptr:
if (p && *p)

**Review by akobrin1 — CHANGES_REQUESTED** [review]
one small correction

**Review by a-ok123 — APPROVED** [review]

**Review by a-ok123 — APPROVED** [review]

**Review by akobrin1 — APPROVED** [review]

**Pull request merged**